test/integration/websocket_integration_test.cc

```diff
@@ -420,7 +422,7 @@ TEST_P(WebsocketIntegrationTest, BidirectionalChunkedData) {
     ASSERT_TRUE(upstream_request_->headers().TransferEncoding() != nullptr);
   }
   if (downstreamProtocol() == Http::CodecClient::Type::HTTP1) {
-    ASSERT_TRUE(response_->headers().TransferEncoding() != nullptr);
+    ASSERT_TRUE(response_->headers().TransferEncoding() == nullptr);
```

Comment:
In theory the upstream is sending an invalid response.  Do you think in the case that an upstream tries to send a chunked response with upgrade we should just stop the T-E or should we reject it as spec-breaking?  Given the remaining output is to send headers without T-E chunked but sending the original chunked body payload I'm inclined to say we should reject the response.

Reply:
Done. I also added a check for non-zero content-length and reject that as well.

I put in support to allow, but strip, `Content-Length: 0` in the spirit of being lenient in what we accept. I think that's probably a pretty common mistake (and indeed Envoy was doing this before my previous change).